raiden/api/python.py

```diff
@@ -865,19 +869,47 @@ def get_token_network_events_raiden(
         )
 
         for block_number, event in raiden_events:
-            new_event = {
-                'block_number': block_number,
-                'event': type(event).__name__,
-            }
-            if hasattr(event, 'transfer'):
-                if event.transfer.token == token_address:
-                    event.transfer = repr(event.transfer)
-            elif hasattr(event, 'token'):
-                if event.token == token_address:
-                    event.balance_proof = repr(event.balance_proof)
-            new_event.update(event.__dict__)
-            returned_events.append(new_event)
+            self._check_and_append_event(returned_events, block_number, event, token_address)
+
         returned_events.sort(key=lambda evt: evt.get('block_number'), reverse=True)
         return returned_events
 
+    def _check_and_append_event(
+            self,
+            returned_events,
+            block_number,
+            event,
+            token_address,
+    ):
+        new_event = {
+            'block_number': block_number,
+            'event': type(event).__name__,
+        }
+        if hasattr(event, 'transfer') and event.transfer.token == token_address:
+            new_event.update(event.__dict__)
+            returned_events.append(new_event)
+        elif hasattr(event, 'token') and event.token == token_address:
```

Comment:
you can use `getattr(event, 'token', None) == token_address` on some of these places

Reply:
Changed it